units/systemd-journal-flush.service

```diff
@@ -18,7 +18,9 @@ RequiresMountsFor=/var/log/journal
 ConditionPathExists=!/etc/initrd-release
 
 [Service]
+ExecStartPre=touch /run/systemd/journal/flushed
 ExecStart=journalctl --flush
+ExecStopPre=rm /run/systemd/journal/flushed
 ExecStop=journalctl --smart-relinquish-var
```

Comment:
That's ugly. Why not create and remove the file from `journalctl` implementation? This makes no difference to operation whether the file is created here or from the binary, and it's generally much better to do a single syscall from the C code instead of having systemd start a child.

Reply:
Those changes look nice in general, but it still feels like the whole thing is more complicated than it needs to be.

Reply:
yes we could also introduce some new options such as `--mark-var-ready` / `--mark-var-gone` and doing `ExecStart=journalctl --mark-var-ready --flush`.

The important part is to make a clear distinction between the "/var/log/journal being ready" event and the flush operation itself.

Reply:
Would that work for you ?